pkg/util/storage.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net/url"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/operator-framework/rukpak/internal/storage"
+	"github.com/operator-framework/rukpak/internal/util"
+	pkgstorage "github.com/operator-framework/rukpak/pkg/storage"
+)
+
+const DefaultBundleCacheDir = "/var/cache/bundles"
+
+func NewBundleStorage(mgr ctrl.Manager, provisionerStorageDirectory, httpExternalAddr, bundleCAFile, bearerToken string) (pkgstorage.Storage, error) {
+	storageURL, err := url.Parse(fmt.Sprintf("%s/bundles/", httpExternalAddr))
```

Comment:
```suggestion
	storageURL, err := url.Parse(fmt.Sprintf("%s/%s/", httpExternalAddr, bundlesPath))
```

We should probably make this configurable because the caller has to know this path when it call the AddExtraHandler function.

Reply:
Oh and we should probably use https://pkg.go.dev/net/url#JoinPath instead of fmt.Sprintf here.

Reply:
Looking at the `url.JoinPath` code it looks like we'd be better suited to do this:
```go
storageURL, err := url.Parse(httpExternalAddr)
...
storageURL.JoinPath(bundlePath)
```

Reply:
I'll make changes.  Thanks!

Reply:
`url.JoinPath` is supported go1.19 or later.  I changed not to use it for now.